fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/observer/eventmappings/AllNodeEventsOneEvent.java

```diff
@@ -66,67 +67,30 @@
     });
 
     @Override
-    public Iterator<FedoraEvent> apply(final Iterator<Event> events) {
-        return new FedoraEventIterator(events);
-    }
-
-    private static class FedoraEventIterator implements Iterator<FedoraEvent> {
-
-        // JCR events in a Multimap keyed by identifier
-        private final Multimap<String, Event> sortedEvents;
-
-        private final Iterator<String> ids;
-
-        public FedoraEventIterator(final Iterator<Event> events) {
-            sortedEvents = index(events, EXTRACT_NODE_ID::apply);
-            ids = sortedEvents.keySet().iterator();
-        }
-
-        @Override
-        public boolean hasNext() {
-            return ids.hasNext();
-        }
-
-        @Override
-        public FedoraEvent next() {
-            final Iterator<Event> nodeSpecificEvents = sortedEvents.get(ids.next()).iterator();
-            // we can safely call next() immediately on nodeSpecificEvents because if there was no event at all, there
-            // would appear no entry in our Multimap under this key
-            final Event firstEvent = nodeSpecificEvents.next();
-
-            final FedoraEvent fedoraEvent = new FedoraEventImpl(firstEvent);
-
-            addProperty(fedoraEvent, firstEvent);
-            while (nodeSpecificEvents.hasNext()) {
-                // add the event type and property name to the event we are building up to emit
-                // we could aggregate other information here if that seems useful
-                final Event otherEvent = nodeSpecificEvents.next();
-                fedoraEvent.addType(FedoraEventImpl.valueOf(otherEvent.getType()));
-                addProperty(fedoraEvent, otherEvent);
-            }
-            return fedoraEvent;
-        }
-
-        @Override
-        public void remove() {
-            // the underlying Multimap is immutable anyway
-            throw new UnsupportedOperationException();
-        }
-
-        private static void addProperty( final FedoraEvent fedoraEvent, final Event ev ) {
-            try {
-                if ( ev.getPath().contains(JCR_CONTENT)) {
-                    fedoraEvent.addProperty("fedora:hasContent");
-                }
-                if (PROPERTY_EVENT_TYPES.contains(ev.getType())) {
-                    final String eventPath = ev.getPath();
-                    fedoraEvent.addProperty(eventPath.substring(eventPath.lastIndexOf('/') + 1));
-                } else {
-                    LOGGER.trace("Not adding non-event property: {}, {}", fedoraEvent, ev);
-                }
-            } catch (final RepositoryException e) {
-                throw new RepositoryRuntimeException(e);
+    public Stream<FedoraEvent> apply(final Stream<Event> events) {
```

Comment:
I believe that this new method is correct, but how about some in-line comments for the poor bastard who has to alter it two years from now? Could be you or me.